### PR TITLE
fix(transformer): class property with typescript value should not be removed

### PR DESCRIPTION
**Pull request by Boshen**

### CODE REVIEW

**Comment by Boshen** [comment]
* **#5298** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5298?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Boshen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Dunqing — APPROVED** [review]

File: tasks/transform_conformance/oxc.snap.md [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
 
 
-# babel-plugin-transform-typescript (2/7)
+# babel-plugin-transform-typescript (2/8)
+* class-property-definition/input.ts
+  x Unresolved references mismatch:
+  | after transform: ["const"]
+  | rebuilt        : []
+
+
```
Comment (Dunqing) [inline tasks/transform_conformance/oxc.snap.md:16]: There seems to be an incorrect check here. The `const` is an unresolved reference, after transformation, the `const` will be removed, Do we need to remove it from `unresolved_references`? @overlookmotel cc

Comment (Boshen) [reply tasks/transform_conformance/oxc.snap.md:16]: I don't know :-)

Comment (overlookmotel) [reply tasks/transform_conformance/oxc.snap.md:16]:
```ts
class Foo {
  #handlers = [] as const;
}
```

`const` in the original TS AST is an `IdentifierReference` which has no binding, so it's recorded in `unresolved_references`. When we remove the `IdentifierReference` in transform, we should remove it from `unresolved_references` too.

**Pull request merged**